File: 2017/day10/part1.py
def solve(data, length):
    values = list(range(length))

    ptr = 0
    skip = 0
    for change in data:
        part = []
        for offset in range(change):
            part.append(values[(ptr + offset) % length])
        part.reverse()
        for offset, val in enumerate(part):
            values[(ptr + offset) % length] = val

        ptr += change + skip
        skip += 1

    return values[0] * values[1]


if __name__ == "__main__":
    assert solve([3, 4, 1, 5], 5) == 12

    with open("input.txt", "r") as file:
        data = list(map(int, file.read().strip().split(",")))
    print(solve(data, 256))

File: 2017/day10/part1.c
#include <stddef.h>
#include <stdio.h>
#include <stdlib.h>
int main(void) {
    FILE *fptr = fopen("input.txt", "r");

    int values[256];
    for (int i = 0; i < 256; i++) {
        values[i] = i;
    }
    char ch;
    int change = 0;
    int skip = 0;
    int j = 0;

    while ((ch = getc(fptr)) != EOF) {
        if (ch == ',' || ch == 10) {
            printf("%i\n", change);
            change = 0;
        } else {
            change *= 10;
            change += ch - '0';
        }
    }

    fclose(fptr);
    return EXIT_SUCCESS;
}
